src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -132,8 +133,17 @@ void CalorimeterHitReco::init() {
     return params;
   };
 
-  auto& serviceSvc = algorithms::ServiceSvc::instance();
-  sampFrac = serviceSvc.service<EvaluatorSvc>("EvaluatorSvc")->compile(m_cfg.sampFrac, hit_to_map);
+  // convert sampFrac string to function that returns sampFrac value
+  try {
+    // trivial function if string is representation of float
+    float sampFrac_value = std::stof(m_cfg.sampFrac);
+    sampFrac = [sampFrac_value](const edm4hep::RawCalorimeterHit&) { return sampFrac_value; };
+  } catch (std::exception& e) {
```

Comment:
This is wrong. `std::stof("1+1")` will succeed to parse `1.0f` from `"1"` and leave out `"+1"`.

Reply:
Good catch. See bab408f.